cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/co/cask/cdap/etl/spark/batch/ETLSpark.java

```diff
@@ -295,4 +303,27 @@ public void destroy() {
     }
   }
 
+  /**
+   * Gets all the plugins used in the program through the {@link PipelinePhase} and calls
+   * {@link #usePluginClass(String, String, String, PluginProperties, PluginSelector)} explicitly to register their
+   * usage so that they are accessible in the current configurer obtained through {@link #getConfigurer()}
+   *
+   * @param phases the {@link PipelinePhase} in this program
+   */
+  private void registerPlugins(PipelinePhase phases) {
```

Comment:
this is repeated a bunch of times. It's better to have a method in PipelinePhase that takes a PluginConfigurer and does all of this. Each program would just call it and pass it getConfigurer()

Reply:
💯 

Done.